common/djangoapps/enrollment/views.py

```diff
@@ -360,11 +362,12 @@ def post(self, request):
         # Check whether any country access rules block the user from enrollment
         # We do this at the view level (rather than the Python API level)
         # because this check requires information about the HTTP request.
+
+        # Lookup the user, instead of using request.user, since request.user may not match the username POSTed.
+        user = User.objects.get(username=username)
```

Comment:
Updated to return HTTP 406 if the user does not exist. Note that this will only occur for server users. Non-server users will get a 404 before we reach this code.